src/main/java/com/google/firebase/auth/FirebaseUserManager.java

```diff
@@ -230,20 +230,28 @@ UserImportResult importUsers(UserImportRequest request) throws FirebaseAuthExcep
 
   Tenant getTenant(String tenantId) throws FirebaseAuthException {
     GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants/" + tenantId);
-    Tenant response = sendRequest("GET", url, null, Tenant.class);
-    if (Strings.isNullOrEmpty(response.getTenantId())) {
-      throw new FirebaseAuthException(TENANT_NOT_FOUND_ERROR, "Failed to get tenant.");
-    }
-    return response;
+    return sendRequest("GET", url, null, Tenant.class);
+  }
+
+  Tenant createTenant(Tenant.CreateRequest request) throws FirebaseAuthException {
+    GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants");
+    url.putAll(request.getProperties());
+    return sendRequest("POST", url, null, Tenant.class);
+  }
+
+  Tenant updateTenant(Tenant.UpdateRequest request) throws FirebaseAuthException {
+    Map<String, Object> properties = request.getProperties();
+    checkArgument(properties.size() > 1,
+        "tenant update must have at least one property other than tenantId set");
+    GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants/" + request.getTenantId());
+    url.putAll(properties);
+    url.putAll(request.getProperties());
```

Comment:
Put the properties in the request body. Remove the extra call to `getProperties()`.

Reply:
Done.